### PR TITLE
snap - fixed param order

### LINKED ISSUE #2916: snap: error: unknown flag `classic'
### Summary

When install neovim via my ansible playbook using the following yaml, I get an error.

### Issue Type

Bug Report

### Component Name

snap

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.11.2]
  config file = /usr/src/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/sbin/ansible
  python version = 3.9.5 (default, May 24 2021, 12:50:35) [GCC 11.1.0]
  jinja version = 3.0.1
  libyaml = True
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
DEFAULT_HOST_LIST(/usr/src/ansible.cfg) = ['/usr/src/hosts.yml']
DEFAULT_REMOTE_USER(/usr/src/ansible.cfg) = david.volm
```

### OS / Environment

Amazon Linux 2

### Steps to Reproduce

`ansible-playbook -vvv playbooks/test.yml -K`
```yaml (paste below)
---
- name: install neovim
  hosts: all
  tasks:
   - name: 'Install neovim via snap'
     become: yes
     community.general.snap:
       name: nvim
       classic: yes
```

### Expected Results

I expect the playbook to install neovim, but instead I get an error.

### Actual Results

```console (paste below)
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_community.general.snap_payload_MrWFp4/ansible_community.general.snap_payload.zip/ansible_collections/community/general/plugins/module_utils/mh/deco.py", line 42, in wrapper
    func(self, *args, **kwargs)
  File "/tmp/ansible_community.general.snap_payload_MrWFp4/ansible_community.general.snap_payload.zip/ansible_collections/community/general/plugins/module_utils/mh/base.py", line 60, in run
    self.__run__()
  File "/tmp/ansible_community.general.snap_payload_MrWFp4/ansible_community.general.snap_payload.zip/ansible_collections/community/general/plugins/module_utils/mh/mixins/state.py", line 36, in __run__
    return func()
  File "/tmp/ansible_community.general.snap_payload_MrWFp4/ansible_community.general.snap_payload.zip/ansible_collections/community/general/plugins/modules/snap.py", line 228, in state_present
ModuleHelperException
fatal: [an_ip_address]: FAILED! => {
    "changed": false,
    "channel": "stable",
    "classic": true,
    "cmd": "['classic', 'channel', 'state', {'actionable_snaps': ['nvim']}]",
    "cmd_args": [
        "/bin/snap",
        "--classic",
        "install",
        "nvim"
    ],
    "force_lang": "C",
    "invocation": {
        "module_args": {
            "channel": "stable",
            "classic": true,
            "name": [
                "nvim"
            ],
            "state": "present"
        }
    },
    "msg": "Ooops! Snap installation failed while executing '['classic', 'channel', 'state', {'actionable_snaps': ['nvim']}]', please examine logs and error output for more details.",
    "output": {
        "channel": "stable",
        "classic": true,
        "cmd": "['classic', 'channel', 'state', {'actionable_snaps': ['nvim']}]",
        "cmd_args": [
            "/bin/snap",
            "--classic",
            "install",
            "nvim"
        ],
        "force_lang": "C",
        "rc": 64,
        "snaps_installed": [
            "nvim"
        ],
        "stderr": "error: unknown flag `classic'\n",
        "stdout": ""
    },
    "rc": 64,
    "snaps_installed": [
        "nvim"
    ],
    "stderr": "error: unknown flag `classic'\n",
    "stderr_lines": [
        "error: unknown flag `classic'"
    ],
    "stdout": "",
    "stdout_lines": [],
    "vars": {
        "channel": "stable",
        "classic": true,
        "cmd": "['classic', 'channel', 'state', {'actionable_snaps': ['nvim']}]",
        "cmd_args": [
            "/bin/snap",
            "--classic",
            "install",
            "nvim"
        ],
        "force_lang": "C",
        "rc": 64,
        "snaps_installed": [
            "nvim"
        ],
        "stderr": "error: unknown flag `classic'\n",
        "stdout": ""
    }
}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Parameter order was wrong.

Fixes #2916 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/packaging/os/snap.py

### CODE REVIEW

**Comment by felixfontein** [comment]
Can you add a changelog fragment?

**Comment by russoz** [comment]
Sure, I was waiting for the tests to pass though.

**Comment by felixfontein** [comment]
@russoz can you rebase and uncomment the new tests in #2920 in this PR?

**Comment by felixfontein** [comment]
I guess the test setup needs to be adjusted to create a symlink from `/snap` to `/var/lib/snapd/snap`.

**Review by aminvakil — APPROVED** [review]
LGTM

File: tests/integration/targets/snap/tasks/default.yml [hunk 1]
```diff
@@ -10,6 +10,12 @@
     name: snapd
     state: started
 
+- name: Create link /snap (RedHat-only)
```
Comment (felixfontein) [inline tests/integration/targets/snap/tasks/default.yml:13]:
```suggestion
- name: Create link /snap
```
This has nothing to do with RedHat :)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks a lot for fixing this!
@aminvakil thanks lot for reviewing!

**Pull request merged**